### PR TITLE
Adds Github Actions as CI

### LINKED ISSUE #19: Add CI
Add Github actions to use Substratee Docker Image 

- [x] Check Build
- [x] Check unit tests
- [x] Add Caching

### PR DESCRIPTION
**Pull request by Gauthamastro**
- Pulls Substratee Docker Image
- Install Rust Toolchain
- Prints Information about the environment
- Builds the project in hardware mode
- Runs `cargo test` in enclave, worker, and client
- Caches Cargo and target folders

### CODE REVIEW

**Comment by murerfel** [comment]
Looks neat! For now this does a build only - do we want to run the tests too, using this Action? Or do you want to have that in a separate Action?

**Comment by Gauthamastro** [comment]
> Looks neat! For now this does a build only - do we want to run the tests too, using this Action? Or do you want to have that in a separate Action?

Yes, I will implement tests shortly

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Install Rust and WASM Environment
+        run: ./ci/install_rust.sh
+      - name: Information
+        run: cargo --version && rustup show && env
+      - name: Build Project
+        run: make
+      - name: Test Client
+        run: cd enclave && cargo test 2>&1
+      - name: Test Worker
+        run: cd workder && cargo test 2>&1
```
Comment (haerdib) [inline .github/workflows/build.yml]: typo, should be worker

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            enclave/target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - name: Install Rust and WASM Environment
+        run: ./ci/install_rust.sh
+      - name: Information
+        run: cargo --version && rustup show && env
+      - name: Build Project
+        run: make
+      - name: Test Client
+        run: cd enclave && cargo test 2>&1
```
Comment (haerdib) [inline .github/workflows/build.yml]: should be client, right?

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Test Worker
+        run: cd workder && cargo test 2>&1
+      - name: Test Enclave
+        run: cd enclave && cargo test 2>&1
```
Comment (haerdib) [inline .github/workflows/build.yml]: will not work. But maybe we can use something similiar as in
 https://github.com/apache/incubator-teaclave-sgx-sdk/issues/232 suggested.

**Review by haerdib — CHANGES_REQUESTED** [review]
We use cargo clippy and fmt in our testing environment. Would this also make sense in this project?

File: .github/workflows/build.yml (hunk 3 above)
Comment (Gauthamastro) [reply .github/workflows/build.yml]: Yes, So this is wrong right? https://github.com/scs/substraTEE-worker/blob/master/Jenkinsfile#L43

Comment (haerdib) [reply .github/workflows/build.yml]: Yes, you're right... we'll need to clean that up

**Review by haerdib — APPROVED** [review]
LGTM

**Pull request merged**